### PR TITLE
migrate to gh actions?

### LINKED ISSUE #48: Switch to GitHub Action instead of Travis
## Current Behavior

We currently use Travis for our CI process.  We'd like to move over to GitHub Actions.  It'll also make our linting process with [golangci-lint](https://golangci-lint.run/usage/install/#github-actions) a little cleaner.

## Expected Behavior

Moving away from Travis for cleaner setup with GitHub Actions.

### PR DESCRIPTION
**Pull request by onyxraven**
Update to use github actions. Along the way this fixes some internal things to do around security and such.

fixes #48 

Versioning
-----

should be none required, but will need a release cut to ensure the release workflow works.

Tasks
-----

* [ ] Manual testing

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v2
+        with:
+          # either 'goreleaser' (default) or 'goreleaser-pro'
+          distribution: goreleaser
+          version: latest
+          args: release --rm-dist
+        env:
+          # this needs push access to the tap
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (onyxraven) [inline .github/workflows/release.yaml]: will add these. this one has to be a PAT to access the other repo :/

Comment (elementalvoid) [reply .github/workflows/release.yaml]: You may need to use a different secret name. `secrets.GITHUB_TOKEN` is a "special" secret that GH Actions auto-creates for every build. I'm not sure what it does is the names conflict.

**Review by elementalvoid — DISMISSED** [review]

**Review by dennis-johnson-dev — DISMISSED** [review]
🚢

**Review by dennis-johnson-dev — APPROVED** [review]
:shipit:

**Pull request merged**